pkg/dump/writer/tar_writer.go

```diff
@@ -87,24 +119,43 @@ func (t *TarWriter) Write(ctx context.Context, object []byte, filePath string) e
 // Flush function flushes all kubernetes object from the buffers to the tar file
 func (t *TarWriter) Flush(ctx context.Context) error {
 	log.I.Debug("Flushing writers")
-	span, _ := tracer.StartSpanFromContext(ctx, span.DumperWriterFlush, tracer.Measured())
-	span.SetTag(tag.DumperWriterTypeTag, TarTypeTag)
-	defer span.Finish()
-	for path, data := range t.buffers {
-		header := &tar.Header{
-			Name: path,
-			Mode: TarWriterChmod,
-			Size: int64(len(*data)),
-		}
+	t.mu.Lock()
+	defer t.mu.Unlock()
 
-		if err := t.tarWriter.WriteHeader(header); err != nil {
+	// Walking the memfs and copying all the files to the tar
+	err := afero.Walk(t.fs, "", func(path string, info os.FileInfo, err error) error {
+		if err != nil {
 			return err
 		}
 
-		if _, err := t.tarWriter.Write(*data); err != nil {
-			return err
+		if info.IsDir() {
+			return nil
+		}
+		file, err := t.fs.Open(path)
+		if err != nil {
+			return fmt.Errorf("open file %s from fs: %w", path, err)
+		}
+		fileInfo, err := t.fs.Stat(path)
+		if err != nil {
+			return fmt.Errorf("stat file %s from fs: %w", path, err)
 		}
-		delete(t.buffers, path)
+		tarHader, err := tar.FileInfoHeader(fileInfo, "")
+		if err != nil {
+			return fmt.Errorf("create tar header for file %s: %w", path, err)
+		}
+		tarHader.Name = path
+		if err := t.tarWriter.WriteHeader(tarHader); err != nil {
+			return fmt.Errorf("write tar header for file %s: %w", path, err)
+		}
```

Comment:
I think i'm confused: does writing a file to a tar archive requires adding an header for each file added individually? (I would have thought that adding a file did that for you but maybe not?)

Also, typo in `tarHader` should be `tarHeader`